File: monitoria_app/lib/app/modules/start/submodulos/alunos/models/aluno.dart
import 'package:cloud_firestore/cloud_firestore.dart';

class Aluno {
  int _id = -1;
  String _nome = '';
  String _email = '';
  String _fase = '';
  String _matricula = '';
  String _curso = '';

  Aluno();

  Aluno.fromJson(String key, Map<dynamic, dynamic> json) {
    this._id = int.parse(key);
    this._curso = json['curso'] ?? '';
    this._email = json['email'] ?? '';
    this._fase = json['fase'] ?? '';
    this._matricula = json['matricula'] ?? '';
    this._nome = json['nome'] ?? '';
  }

  Map<String, dynamic> toJsonUpdate() {
    Map<dynamic, dynamic> json = Map<dynamic, dynamic>();
    json['curso'] = this._curso;
    json['nome'] = this._nome;
    json['fase'] = this._fase;
    json['matricula'] = this._matricula;
    json['email'] = this._email;
    return {this._id.toString(): json};
  }

  Map<String, dynamic> toJsonAdd() {
    Map<String, dynamic> json = Map<String, dynamic>();
    json['curso'] = this._curso;
    json['nome'] = this._nome;
    json['fase'] = this._fase;
    json['matricula'] = this._matricula;
    json['email'] = this._email;
    return json;
  }

  int get id => _id;

  set id(int value) => _id = value;

  String get nome => _nome;

  set nome(String value) => _nome = value;

  String get email => _email;

  set email(String value) => _email = value;

  String get fase => _fase;

  set fase(String value) => _fase = value;

  String get matricula => _matricula;

  set matricula(String value) => _matricula = value;

  String get curso => _curso;

  set curso(String value) => _curso = value;

  @override
  String toString() {
    return 'Nome: $_nome';
  }
}

File: gerador/models/Aluno.py
class Aluno:

    def __init__(self):
        self.__id = ''
        self.__nome = ''
        self.__matricula = ''
        self.__email = ''
        self.__fase = ''
        self.__curso = ''

    def id(self, id):
        self.__id = id
        return self

    def getId(self):
        return self.__id

    def nome(self, nome):
        self.__nome = nome
        return self

    def matricula(self, matricula):
        self.__matricula = matricula
        return self

    def email(self, email):
        self.__email = email
        return self

    def curso(self, curso):
        self.__curso = curso
        return self

    def fase(self, fase):
        self.__fase = fase
        return self

    def getNome(self):
        return self.__nome

    def __str__(self):
        aluno = '('
        aluno += self.__id
        aluno += ') - '
        aluno += 'Nome: '
        aluno += self.__nome
        return aluno

    def getInfo(self):
        aluno = 'Nome: '
        aluno += self.__nome
        aluno += '\nMatricula: '
        aluno += self.__matricula
        aluno += '\nEmail: '
        aluno += self.__email
        aluno += '\nCurso: '
        aluno += self.__curso
        aluno += '\nFase: '
        aluno += self.__fase
        return aluno

    def toJson(self):
        json = {}
        json['nome'] = self.__nome
        json['matricula'] = self.__matricula
        json['email'] = self.__email
        json['curso'] = self.__curso
        json['fase'] = self.__fase
        return json

    def fromJson(self, json, id):
        if('nome' in json):
            self.__nome = json['nome']
        if('matricula' in json):
            self.__matricula = json['matricula']
        if('email' in json):
            self.__email = json['email']
        if('curso' in json):
            self.__curso = json['curso']
        if('fase' in json):
            self.__fase = json['fase']
        self.__id = id

    def copy(self):
        a = Aluno()
        a.nome(self.__nome)
        a.curso(self.__curso)
        a.email(self.__email)
        a.fase(self.__fase)
        a.id(self.__id)
        a.matricula(self.__matricula)
        return a

    def toList(self):
        aluno = []
        aluno.append(self.__nome)
        aluno.append(self.__matricula)
        aluno.append(self.__email)
        aluno.append(self.__curso)
        aluno.append(self.__fase)
        return list(aluno)
